projects/plugins/jetpack/_inc/lib/tonesque.php

```diff
@@ -72,21 +72,24 @@ public static function imagecreatefromurl( $image_url ) {
 			// If it's a url pointing to a local media library url.
 			$content_url = content_url();
 			$_image_url  = set_url_scheme( $image_url );
-			if ( wp_startswith( $_image_url, $content_url ) ) {
+			if ( str_starts_with( $_image_url, $content_url ) ) {
 				$_image_path = str_replace( $content_url, WP_CONTENT_DIR, $_image_url );
 				if ( file_exists( $_image_path ) ) {
 					$filetype = wp_check_filetype( $_image_path );
 					$type     = $filetype['type'];
 
-					if ( wp_startswith( $type, 'image/' ) ) {
+					if ( str_starts_with( $type, 'image/' ) ) {
 						$data = file_get_contents( $_image_path ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
 					}
 				}
 			}
 
 			if ( empty( $data ) ) {
 				$response = wp_safe_remote_get( $image_url );
-				if ( is_wp_error( $response ) ) {
+				if (
+					is_wp_error( $response )
+					|| ! str_starts_with( $response['headers']['content-type'], 'image/' )
```

Comment:
That would definitely be an issue.  Are you aware of any sites/services that do that?

Reply:
I don't know of any offhand, but the `$response['headers']['content-type']` will have whatever value the remote server decides to return for the content type, if it specifies a type at all.

Most servers will probably get it right, but some may not for some image types. For example, if someone's webserver is old enough it may not know about `image/webp` to call webp images that type.

Reply:
Do you have a preference on an alternative?  My goal is to avoid calling `imagecreatefromstring()` on something that isn't a function, and am happy to try out other approaches.

Reply:
Really calling `imagecreatefromstring()` (maybe with `@` and manual error handling) is probably the most straightforward way to check if you actually got an image.

Detecting an error page would be better done by looking at the response code. If the server returned a 2xx code, it's probably not an error page, and otherwise it probably is.

If you really want to check the type, you'd have to use something like `jetpack_mime_content_type()` except every function I see for doing something like that (including that one) wants to be passed a filename rather than the contents of the file.

Reply:
Would something like this work for you?
493296901735d18c7e2d26c94af137c568401bf2

Reply:
> Hiding things behind `@` is generally discouraged.

There's a difference between using `@` to ignore errors and using `@` to suppress PHP's default warnings in order to handle the problem in a better manner. But I'm fine with the response code check here, it's still an improvement.